src/ffi/http_types.rs

```diff
@@ -428,6 +435,37 @@ ffi_fn! {
     }
 }
 
+ffi_fn! {
+    /// Iterates the headers in the order the were recieved, passing each name and value pair to the callback.
+    ///
+    /// The `userdata` pointer is also passed to the callback.
+    ///
+    /// The callback should return `HYPER_ITER_CONTINUE` to keep iterating, or
+    /// `HYPER_ITER_BREAK` to stop.
+    fn hyper_headers_foreach_ordered(headers: *const hyper_headers, func: hyper_headers_foreach_callback, userdata: *mut c_void) {
```

Comment:
Should I add some logic here along the lines of:
```
If !header_order_preserved { // or headers.orig_order is default
   hyper_headers_foreach(headers, func, userdata);
   return;
}
```
Currently, if the connection does not have `preserve_header_order` set, this function will iterate over nothing.

Reply:
I think probably it should all just be one function. If the order map is set, then use it, otherwise iterate normally.

Reply:
done